extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaConfiguration.java

```diff
@@ -12,11 +12,36 @@
 @ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
 public class TikaConfiguration {
     /**
-     * The path to the tika-config.xml
+     * Path to the tika-config.xml
      */
     @ConfigItem
     public Optional<String> tikaConfigPath;
 
+    /**
+     * Comma separated list of the parsers which must be supported.
+     * 
+     * Most of the document formats recognized by Apache Tika are supported by default but it affects
+     * the application memory and native executable sizes. One can list only the required parsers in
+     * tika-config.xml to minimize a number of parsers loaded into the memory, but using this
+     * property is recommended to achieve both optimizations.
+     * 
+     * Either the abbreviated or full parser class names can be used.
+     * At the moment only PDF parser can be listed using a reserved 'pdf' abbreviation.
+     * Using the custom abbreviations is recommended for all other parsers.
+     * 
+     * <pre>
+     * // Only PDF parser is required:
+     * tika-parsers = pdf
+     * // Only PDF and Java class parsers are required:
+     * tika-parsers = pdf,classparser
+     * classparser=org.apache.tika.parser.asm.ClassParser
+     * </pre>
+     * 
+     * This property will have no effect if the `tikaConfigPath' property has been set.
+     */
+    @ConfigItem
+    public Optional<String> parsers;
```

Comment:
If this is a list, shouldn't it be a `List<String>`? Also, if more parsers are supported by default, shouldn't there be a default value listing the supported parsers?

If the set of possible parsers is fixed, then rather than giving class names, it would be better to make an `enum` with the exact available choices.

Reply:
@dmlloyd 
This property instructs the extension which parser service providers should be loaded/added for the reflection, so I'd not like the extension 'committing' to some default values because if the users may need for ex PDF, but only Microsoft Office formats. 

The other problem is that a `tika-parsers` dependency brings 68 parsers with it, so it is difficult to decide which parsers will be supported OOB at the dependency level until after a [modularization](https://issues.apache.org/jira/browse/TIKA-2943) task is complete.
My plan has been to enhance this list of abbreviations such as `pdf` based on the demand. We start from `pdf`. If users say, why do we need to type a class name for this parser, then we make a new abbreviation recognized OOB. 
Keeping the list of parsers open-ended and use String may work better for it, given that the custom abbreviations are supported, see my comment about it above where you spotted the docs typo :-).

Reply:
Sounds good.